### ISSUE TITLE
project-local MRU

### ISSUE THREAD

**Issue by Konfekt** [yegappan/mru#67]
For what it's worth, similar to https://gist.github.com/Konfekt/b70fcfa4a3c83762d69fe9bb6c2af3d5 a snippet for a project-local MRU

```vim
" - Default to global mru if no local mru found
" - start with global mru if local mru is empty
" - Make Sessions respect local mru
"  Add .mru to to .git/info/exclude (maybe best in the git template) to avoid clutter

if empty($XDG_STATE_HOME)
  let $XDG_STATE_HOME = $HOME . '/.local/state'
endif
if !isdirectory(expand($XDG_STATE_HOME))
  call mkdir(expand($XDG_STATE_HOME), 'p')
endif

function s:set_local_mru()
  let g:MRU_File = $XDG_STATE_HOME . '/vim/mru'
  let g:local_mru_file = findfile('.mru', '.;')
  if empty(g:local_mru_file)
    let g:local_mru_file = g:MRU_File
  elseif getfsize(g:local_mru_file) == 0 && exists('*filecopy')
    call delete(g:local_mru_file)
    call filecopy(g:MRU_File, g:local_mru_file)
  endif
  let g:MRU_File = g:local_mru_file
endfunction

call s:set_local_mru()
autocmd vimrc SessionLoadPost * call s:set_local_mru()
```

**Comment by yegappan** [comment]
The g:MRU_File variable will be set outside of the plugin to either the global MRU file name or to a project-local file name?  You expect the plugin to always use the latest value of the g:MRU_File variable?

**Comment by Konfekt** [comment]
> The g:MRU_File variable will be set outside of the plugin to either the global MRU file name or to a project-local file name? 

Yes, that's the idea. 
Similar to `oldfiles` in `.viminfo` (as a specific effect of the local-viminfo plugin).
The snippet can be added as is and loads any `.mru` already fine.
It could be documented, say in the Readme, doc or wiki, or included as an opt-in.

> You expect the plugin to always use the latest value of the g:MRU_File variable?

I somehow wanted the `.viminfo` to respect sessions, and I tried this with this plug-in as well. 
That it can't change afterwards is mainly relevant if one uses sessions and loads another one in a running Vim instance.

**Comment by mmrwoods** [comment]
`command! MRUCwd execute 'MRU ' . getcwd()` ?

**Comment by Konfekt** [comment]
Yes, that will work always if that command will be used. The idea that came with `local-viminfo` is that after switching projects, say by loading a session, besides the local viminfo also other commands such as `MRU` will default to the new project root

**Comment by Konfekt** [comment]
In practice it never bothered me, it was just reasoning on general principles. I use `MRU` a lot

**Comment by mmrwoods** [comment]
Oh I see, so a change to MRU to always read `g:MRU_File` at runtime rather than load time would allow code external to MRU to set is as needed. 

Maybe it would be easier to add a function to allow `s:MRU_File` to be set, e.g.

```vim
func MruSetFile(path)
  let s:MRU_File = expand(a:path)
endfunc
```

Sorry btw for jumping on this thread, I just spotted it while looking for something else, and I was curious.

**Comment by Konfekt** [comment]
Your interest is much appreciated. Yes, that sounds like a straight-forward solution. Question is on maintainer's side if people care enough. If you do, maybe that would verge on a critical mass
